### PR TITLE
Fixed the SMTP password configuration regression.

### PR DESCRIPTION
**Pull request by bpmcd**
Not setting a SMTP password in CM will cause the alert engine to skip SMTP authc.

### CODE REVIEW

**Comment by bpmcd** [comment]
#50

File: alert-engine/src/test/scala/io/phdata/pulse/alertengine/AlertEngineCliParserTest.scala [hunk 1]
```diff
@@ -23,14 +23,22 @@ class AlertEngineCliParserTest extends FunSuite {
...
-  def setEnv(key: String, value: String) = {
+  def setEnv(key: String, value: String): String = {
     val field = System.getenv().getClass.getDeclaredField("m")
     field.setAccessible(true)
     val map =
       field.get(System.getenv()).asInstanceOf[java.util.Map[java.lang.String, java.lang.String]]
     map.put(key, value)
   }
 
+  def unsetEnv(key: String): String = {
```
Comment (afoerster) [inline alert-engine/src/test/scala/io/phdata/pulse/alertengine/AlertEngineCliParserTest.scala]: Is this function used somewhere other than a test?

Comment (bpmcd) [reply alert-engine/src/test/scala/io/phdata/pulse/alertengine/AlertEngineCliParserTest.scala]: No.

**Comment by bpmcd** [comment]
Some environments do not require an SMTP password.  With the current implementation Alert-engine cannot support that configuration.  It worked in v1.x, but is now not working in v2.x.

**Pull request merged**